chia/server/server.py

```diff
@@ -157,15 +164,47 @@ def __init__(
         if self._local_type is NodeType.INTRODUCER:
             self.introducer_peers = IntroducerPeers()
 
-        if self._local_type is not NodeType.INTRODUCER:
-            self._private_cert_path, self._private_key_path = private_ssl_paths(root_path, config)
-        if self._local_type is not NodeType.HARVESTER:
-            self.p2p_crt_path, self.p2p_key_path = public_ssl_paths(root_path, config)
+        ca_private_crt_path, ca_private_key_path = private_ca_crt_key
+        chia_ca_crt_path, chia_ca_key_path = chia_ca_crt_key
+
+        private_cert_path, private_key_path = None, None
+        public_cert_path, public_key_path = None, None
+
+        authenticated_client_types = {NodeType.HARVESTER}
+        authenticated_server_types = {NodeType.HARVESTER, NodeType.FARMER, NodeType.WALLET, NodeType.DATA_LAYER}
+
+        if self._local_type in authenticated_client_types:
+            # Authenticated clients
+            private_cert_path, private_key_path = private_ssl_paths(root_path, config)
+            self.ssl_client_context = ssl_context_for_client(
+                ca_private_crt_path, ca_private_key_path, private_cert_path, private_key_path
+            )
+        else:
+            # Public clients
+            public_cert_path, public_key_path = public_ssl_paths(root_path, config)
+            self.ssl_client_context = ssl_context_for_client(
+                chia_ca_crt_path, chia_ca_key_path, public_cert_path, public_key_path
+            )
+
+        if self._local_type in authenticated_server_types:
+            # Authenticated servers
+            private_cert_path, private_key_path = private_ssl_paths(root_path, config)
+            self.ssl_context = ssl_context_for_server(
+                ca_private_crt_path,
+                ca_private_key_path,
+                private_cert_path,
+                private_key_path,
+                log=self.log,
+            )
         else:
-            self.p2p_crt_path, self.p2p_key_path = None, None
-        self.ca_private_crt_path, self.ca_private_key_path = private_ca_crt_key
-        self.chia_ca_crt_path, self.chia_ca_key_path = chia_ca_crt_key
-        self.node_id = self.my_id()
+            # Public servers
+            public_cert_path, public_key_path = public_ssl_paths(root_path, config)
+            self.ssl_context = ssl_context_for_server(
+                chia_ca_crt_path, chia_ca_key_path, public_cert_path, public_key_path, log=self.log
+            )
+
+        # If node has public cert use that one for id, if not use private.
+        self.node_id = calculate_node_id(private_cert_path if public_cert_path is None else public_cert_path)
```

Comment:
Should probably apply https://gist.github.com/altendky/2d316b1e93f99352868f73623bb85ce3 (or I can PR separately).  It points out that `private_cert_path` is `Optional` at this point when expected to be a `Path`.

Reply:
The situation here is non-trivial but there doesn't appear to be an actual bug.  Hinting can be dealt with later.  In short, `self._local_type` is always in either `authenticated_client_types` or `authenticated_server_types` (`private_cert_path` is not `None`), or it is in neither (`public_cert_path` is not `None`).